Python/Product/Analysis/Interpreter/Ast/AstAnalysisFunctionWalker.cs

```diff
@@ -25,25 +25,20 @@ namespace Microsoft.PythonTools.Interpreter.Ast {
     class AstAnalysisFunctionWalker : PythonWalker {
         private readonly FunctionDefinition _target;
         private readonly NameLookupContext _scope;
-        private readonly List<IPythonType> _returnTypes;
         private readonly AstPythonFunctionOverload _overload;
         private AstPythonType _selfType;
 
         public AstAnalysisFunctionWalker(
             NameLookupContext scope,
-            FunctionDefinition targetFunction
+            FunctionDefinition targetFunction,
+            AstPythonFunctionOverload overload
         ) {
             _scope = scope ?? throw new ArgumentNullException(nameof(scope));
             _target = targetFunction ?? throw new ArgumentNullException(nameof(targetFunction));
-            _returnTypes = new List<IPythonType>();
-            _overload = new AstPythonFunctionOverload(
-                AstPythonFunction.MakeParameters(_scope.Ast, _target),
-                _scope.GetLocOfName(_target, _target.NameExpression),
-                _returnTypes
-            );
+            _overload = overload;
         }
 
-        public IList<IPythonType> ReturnTypes => _returnTypes;
+        public IList<IPythonType> ReturnTypes => _overload.ReturnTypes;
```

Comment:
according to the constructor, `_overload` can be `null`.

Reply:
it's used in a few other places without checks, so I suppose ctor should throw if null.